mlflow/ml-package-versions.yml

```diff
@@ -337,3 +337,15 @@ spark:
     run: |
       # TODO: Add datasource autologging tests
       find tests/spark_autologging/ml -name 'test*.py' | xargs -L 1 pytest --large
+
+pip:
+  package_info:
+    pip_release: "pip"
+    install_dev: |
+      pip install git+https://github.com/pypa/pip.git
+
+  models:
+    minimum: "20.1"
```

Comment:
In pip >= 20.1, `pip._internal.req.parse_requirements` returns instances of `ParsedRequirement` which provides an attribute named `requirement` which stores a raw requirement string.

In pip < 20.1, `pip._internal.req.parse_requirements` returns instances of `InstallRequirement` which doesn't provide such attribute. It's a bit cumbersome to restore the raw requirement string.

Reply:
Todo: Check the version of pip installed on MLRs

Reply:
Pip versions in MLRs:

| MLR | pip version |
| :-- | :---------- |
| 7.6 | 20.0.2      |
| 8.0 | 20.2.4      |
| 8.1 | 20.2.4      |
| 8.2 | 20.2.4      |

Source: https://docs.databricks.com/runtime/mlruntime.html#libraries-included-in-databricks-runtime-ml

Reply:
Out of curiosity, what did older versions of conda do to support this?

Reply:
Conda just does the following when installing pip dependencies:

1. Store pip dependencies as a temporary requirements file.
2. And then, run `pip install -r` against the temporary file using `subprocess`.

https://github.com/conda/conda/blob/33a142c16530fcdada6c377486f1c1a385738a96/conda_env/installers/pip.py#L53

So conda doesn't have to think about how `pip._internal.req.parse_requirements` behaves.